src/packet_distribution.cpp

```diff
@@ -470,27 +506,69 @@ void AOApplication::server_packet_received(AOPacket *p_packet)
   }
   else if (header == "MS") {
     if (courtroom_constructed && courtroom_loaded)
+    {
       w_courtroom->handle_chatmessage(&p_packet->get_contents());
+      if (get_auto_logging_enabled() && !log_filename.isEmpty())
```

Comment:
Reduce redundancy in all of these code blocks. You know the ones. Put it in a function or something.

Reply:
ya wtf cw i thought you were better than this

Reply:
ae